pkg/resources/notification_integration.go

```diff
@@ -0,0 +1,291 @@
+
+		Schema: notificationIntegrationSchema,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+	}
+}
+
+// CreateNotificationIntegration implements schema.CreateFunc
+func CreateNotificationIntegration(d *schema.ResourceData, meta interface{}) error {
+	db := meta.(*sql.DB)
+	name := d.Get("name").(string)
+
+	stmt := snowflake.NotificationIntegration(name).Create()
+
+	// Set required fields
+	stmt.SetString(`TYPE`, d.Get("type").(string))
+	stmt.SetBool(`ENABLED`, d.Get("enabled").(bool))
+
+	// Set optional fields
+	if v, ok := d.GetOk("comment"); ok {
+		stmt.SetString(`COMMENT`, v.(string))
+	}
+
+	// Now, set the notification provider
+	err := setNotificationProviderSettings(d, stmt)
+	if err != nil {
+		return err
+	}
+
+	err = snowflake.Exec(db, stmt.Statement())
+	if err != nil {
+		return fmt.Errorf("error creating notification integration: %w", err)
+	}
+
+	d.SetId(name)
+
+	return ReadNotificationIntegration(d, meta)
+}
+
+// ReadNotificationIntegration implements schema.ReadFunc
+func ReadNotificationIntegration(d *schema.ResourceData, meta interface{}) error {
+	db := meta.(*sql.DB)
+	id := d.Id()
+
+	stmt := snowflake.NotificationIntegration(d.Id()).Show()
+	row := snowflake.QueryRow(db, stmt)
+
+	// Some properties can come from the SHOW INTEGRATION call
+
+	s, err := snowflake.ScanNotificationIntegration(row)
+	if err != nil {
+		return fmt.Errorf("Could not show notification integration: %w", err)
+	}
+
+	// Note: category must be NOTIFICATION or something is broken
+	if c := s.Category.String; c != "NOTIFICATION" {
+		return fmt.Errorf("Expected %v to be a NOTIFICATION integration, got %v", id, c)
+	}
+
+	if err := d.Set("name", s.Name.String); err != nil {
+		return err
+	}
+
+	if err := d.Set("type", s.IntegrationType.String); err != nil {
+		return err
+	}
+
+	if err := d.Set("created_on", s.CreatedOn.String); err != nil {
+		return err
+	}
+
+	if err := d.Set("enabled", s.Enabled.Bool); err != nil {
+		return err
+	}
+
+	// Some properties come from the DESCRIBE INTEGRATION call
+	// We need to grab them in a loop
+	var k, pType string
+	var v, unused interface{}
+	stmt = snowflake.NotificationIntegration(d.Id()).Describe()
+	rows, err := db.Query(stmt)
+	if err != nil {
+		return fmt.Errorf("Could not describe notification integration: %w", err)
+	}
+	defer rows.Close()
+	for rows.Next() {
+		if err := rows.Scan(&k, &pType, &v, &unused); err != nil {
+			return err
+		}
+		switch k {
+		case "ENABLED":
+			// We set this using the SHOW INTEGRATION call so let's ignore it here
+
+		case "GCP_PUBSUB_SUBSCRIPTION_NAME":
+			if err = d.Set("gcp_pubsub_subscription_name", v.(string)); err != nil {
+				return err
+			}
+		case "GCP_PUBSUB_SERVICE_ACCOUNT":
+			if err = d.Set("gcp_pubsub_service_account", v.(string)); err != nil {
+				return err
+			}
+
+		case "AZURE_CONSENT_URL":
+			if err = d.Set("azure_consent_url", v.(string)); err != nil {
```

Comment:
you forgot to put `azure_consent_url` and `azure_multi_tenant_app_name` in the `notificationIntegrationSchema`